### PR TITLE
feat: launchpad changes

### PR DESCRIPTION
**Pull request by RolandSherwin**
- Revamps the TUI to use the new color pallet
- Adds support for displaying some stats
- Reset the services if the discord id changes

This PR will be followed up with more changes to polish things up.

### CODE REVIEW

File: node-launchpad/src/components/footer.rs [hunk 1]
```diff
@@ -6,50 +6,126 @@
...
+            Style::default().fg(GHOST_WHITE)
+        } else {
+            Style::default().fg(LIGHT_PERIWINKLE)
+        };
+
+        let (line1, line2) = match self.current_scene {
             Scene::Home => {
-                "[Ctrl+g] Start nodes, [Ctrl+x] Stop nodes, [O] Set Resources, [D]iscord Username, [Q]uit".into()
+                let line1 = Line::from(vec![
+                    Span::styled(" [Ctrl+G] ", command_style),
```
Comment (b-zee) [inline node-launchpad/src/components/footer.rs:86]: Is there a reason for the space in here?

File: node-launchpad/src/components/resource_allocation.rs [hunk 2]
```diff
@@ -21,7 +21,8 @@ use crate::{
 use super::{utils::centered_rect_fixed, Component};
 
 pub const GB_PER_NODE: usize = 5;
-pub const GB: usize = 1024 * 1024 * 1024;
+pub const MB: usize = 1000 * 1000;
+pub const GB: usize = MB * 1000;
```
Comment (b-zee) [inline node-launchpad/src/components/resource_allocation.rs:25]: Just a nitpick, but I think we're using `MiB`/`GiB` (mebibyte/gibibyte) mostly in our apps (1024/binary based), lining up with chunks which are 512KiB(?).

**Comment by b-zee** [comment]
TUI looks clean with those colors, nice job!

File: node-launchpad/src/components/footer.rs [hunk 3]
```diff
@@ -6,50 +6,126 @@
...
+                    Span::styled(
+                        "Manage Nodes               ",
+                        Style::default().fg(EUCALYPTUS),
+                    ),
+                    Span::styled("[D] ", Style::default().fg(GHOST_WHITE)),
+                    Span::styled(
+                        "Beta Rewards Programme      ",
+                        Style::default().fg(VERY_LIGHT_AZURE),
+                    ),
+                    Span::styled("[Q] ", Style::default().fg(GHOST_WHITE)),
```
Comment (b-zee) [inline node-launchpad/src/components/footer.rs:105]: Maybe use lower case `o`, `d` and `q` here. Might be just me being pedantic, but I pressed `Shift` a few times.

Just looked at `dua`, which is quite a popular app using `ratatui`: https://github.com/Byron/dua-cli/blob/ced3b4f5e375278dbee52319eac8750b14eb328a/src/interactive/widgets/help.rs#L43

File: node-launchpad/src/components/footer.rs (hunk 1 above)
Comment (RolandSherwin) [reply node-launchpad/src/components/footer.rs:86]: Those spaces are to accomodate the borders. I haven't found a clean way to create a border, and then split up the remaining space into a `Layout` yet. 
It is scattered all over the codebase, and I'm not proud of it, but I don't have any alternate route yet.

File: node-launchpad/src/components/resource_allocation.rs (hunk 2 above)
Comment (RolandSherwin) [reply node-launchpad/src/components/resource_allocation.rs:25]: I used `MiB` earlier but here this is for fs storage and I crosschecked the space used by a folder with the ones from macos -> right click -> info and they did not match. So I assume we should be using `MB/GB` for fs?

File: node-launchpad/src/components/footer.rs (hunk 3 above)
Comment (RolandSherwin) [reply node-launchpad/src/components/footer.rs:105]: The design is based of off a desgin doc shared on Slack, probably Jim is the guy you'd want to discuss this with.

Comment (jacderida) [reply node-launchpad/src/components/footer.rs:105]: Yeah, I've also pointed out the lowercase/uppercase discrepancy a few times. I gravitated towards using shift too.

**Comment by joshuef** [comment]
Screenshot 2024-05-31 at 09 27 38

**Comment by joshuef** [comment]
Can we drop the bg colour? It seems problematic to me

**Comment by joshuef** [comment]
Screenshot 2024-05-31 at 09 28 48

this is via baked in terminal, looks better

**Pull request closed without merging**